src/transaction.js

```diff
@@ -69,6 +97,12 @@ Transaction.fromBuffer = function (buffer, __noStrict) {
     })
   }
 
+  if (hasWitnesses) {
+    for (i = 0; i < vinLen; ++i) {
+      tx.inputs[i].witness = readScript()
```

Comment:
Sorry, the number of witnesses is implicit. The rest is fine: you know there are `ninputs` witnesses. the first character you read is the number of elements for that witness. The first witness has zero elements, so it didn't use segwit. The second would be a P2PKH embedded in Segwit.

I'll whip up something to parse them to arrays